From a7cea3d69b3b3577e661826e4f20596ab64c6847 Mon Sep 17 00:00:00 2001
From: Alison Wang <alison.wang@nxp.com>
Date: Thu, 19 Jul 2018 17:45:36 +0800
Subject: [PATCH 277/706] drm: hdp: Set the Lane link rate according to link
 training

This patch sets the Lane link rate for CDN_API_DPTX_Set_VIC_blocking
according to the result of link training.

Signed-off-by: Alison Wang <alison.wang@nxp.com>
(cherry picked from commit f10a3c283f58be04f00c798d00d0d31c540252e7)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/gpu/drm/imx/hdp/imx-dp.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/imx/hdp/imx-dp.c b/drivers/gpu/drm/imx/hdp/imx-dp.c
index ad309f90189a..2d4dd5ba8fc9 100644
--- a/drivers/gpu/drm/imx/hdp/imx-dp.c
+++ b/drivers/gpu/drm/imx/hdp/imx-dp.c
@@ -180,7 +180,7 @@ void dp_mode_set(state_struct *state, int vic, int format, int color_depth, int
 	       rls.swing[0], rls.swing[1], rls.swing[2],
 	       rls.preemphasis[0], rls.preemphasis[1], rls.preemphasis[2]);
 
-	switch (max_link_rate) {
+	switch (rls.rate) {
 	case 0x0a:
 		sym_rate = RATE_2_7;
 		break;
-- 
2.17.1

